include/mesh/distributed_mesh.h

```diff
@@ -89,6 +89,12 @@ class DistributedMesh : public UnstructuredMesh
   virtual std::unique_ptr<MeshBase> clone () const override
   { return libmesh_make_unique<DistributedMesh>(*this); }
 
+  /**
+   * Assignment function, will move argument onto the MeshBase object.
+   * Definition in derived classes.
+   */
+  virtual void assign (MeshBase & other_mesh) override;
+
```

Comment:
But you aren't implementing "move" assignment, you're implementing (non-standard) copy assignment.

Reply:
Vikram can correct me if I'm wrong, but I think what he wants here is actual move assignment, not copy assignment.  A swap() might work as an alternative, or as a better alternative for other use cases further down the line?

IIRC putting this into a "named assignment" was my idea, to try to make it harder for users to accidentally call an expensive copy.  If I'm right about move-and-later-swap being the right goals, though, then perhaps "letting C++ give users extra rope" is a worthwhile price to pay for letting C++ do its other magic?  IIRC with C++11 and later, if you make a class move constructable and move assignable, the standard then provides you with a std::swap implementation automatically.  And if we turned this into a proper move assignment operator that'd be most of the work done for when we want to add a move constructor later.